linode/manager
# feat: [M3-7643] – Support VPC IPv4 Ranges in Linode Create flow

## Description 📝
Add support for VPC IPv4 Ranges in the Linode Create flow.

- "Add additional IPv4 ranges" section added in VPC panel
- Logic to include `ip_ranges` in creation payload

The inclusion of a "VPC IPv4 Ranges" column in the VPC Detail > Subnets inner table as per the mockups slipped through the cracks during ticket creation, but I noticed it and included it in this PR (to try to keep things more consolidated).

## Changes  🔄


## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-01-26 at 12 31 24 PM](https://github.com/linode/manager/assets/114682940/d1936691-6af3-4693-9ce3-1508fd3dd6d1) | ![Screenshot 2024-01-26 at 12 37 46 PM](https://github.com/linode/manager/assets/114682940/8d9ddd98-6792-4035-9327-1025c01a68af) |
![Screenshot 2024-01-26 at 12 55 08 PM](https://github.com/linode/manager/assets/114682940/39d02c1d-9c85-4f85-b518-0372e310361a) | ![Screenshot 2024-01-26 at 12 54 12 PM](https://github.com/linode/manager/assets/114682940/7e90ec8b-e54d-4733-a6c4-0dc92799e970)


## How to test 🧪
### Verification steps 
With the `vpc-extra-beta` tag on your account, enter the Linode Create flow and confirm:

- The "Add additional IPv4 ranges" section appears in the VPC panel after a subnet is selected
- You can add and remove IP input fields using the "Add IPv4 Range" and the "X" buttons, respectively
- When you click "Create Linode," the POST request contains the IP ranges you specified in the form in the `ip_ranges` property
- Once the linode is created, go to the detail page of the VPC you assigned the linode to. Expand the inner subnet table and observe the "VPC IPv4 Ranges" column is populated with what you provided (or "None" if you didn't provide any).

## As an Author I have considered 🤔
